javav2/example_code/ecr/src/main/java/com/example/ecr/scenario/ECRScenario.java

```diff
@@ -0,0 +1,335 @@
+        If the push operation is successful, the method prints a message indicating that the image was pushed to ECR.
+        """);
+        waitForInputToContinue(scanner);
+
+        try {
+            ecrActions.pushDockerImage(repoName, localImageName);
+
+        } catch (RuntimeException e) {
+            System.err.println("An error occurred while pushing a local Docker image to Amazon ECR: " + e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("8. Verify if the image is in the ECR Repository.");
+        waitForInputToContinue(scanner);
+        try {
+            ecrActions.verifyImage(repoName, localImageName);
+        } catch (RuntimeException e) {
+            System.err.println("An error occurred while pushing a local Docker image to Amazon ECR: " + e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("9. As an optional step, you can interact with the image in Amazon ECR by using the CLI.");
+        System.out.println("Would you like to view instructions on how to use the CLI to run the image? (y/n)");
+        String ans = scanner.nextLine().trim();
+        if (ans.equalsIgnoreCase("y")) {
+            String instructions = """
+            1. Authenticate with ECR - Before you can pull the image from Amazon ECR, you need to authenticate with the registry. You can do this using the AWS CLI:
+
+                aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin %s.dkr.ecr.us-east-1.amazonaws.com
+
+            2. Describe the image using this command:
+
+               aws ecr describe-images --repository-name %s --image-ids imageTag=%s
+
+            3. Run the Docker container and view the output using this command:
+
+               docker run --rm %s.dkr.ecr.us-east-1.amazonaws.com/%s:%s
+            """;
+
+            instructions = String.format(instructions, accountId, repoName, localImageName, accountId, repoName, localImageName);
+            System.out.println(instructions);
+        }
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("10. Delete the ECR Repository.");
+        System.out.println(
+        """
+        If the repository isn't empty, you must either delete the contents of the repository 
+        or use the force option (used in this scenario) to delete the repository and have Amazon ECR delete all of its contents 
+        on your behalf.
+        """);
+        System.out.println("Would you like to delete the Amazon ECR Repository? (y/n)");
+        String delAns = scanner.nextLine().trim();
+        if (delAns.equalsIgnoreCase("y")) {
```

Comment:
It's still possible for the ECR repo to be created but not destroyed because an error happened in between. When I do this I like to have separate steps for `deploy` `run` and `destroy`. That way the the user can run the clean up steps independently of the scenario.

Reply:
Addressed this comment.